lib/ansible/modules/crypto/acme/acme_challenge_cert_helper.py

```diff
@@ -206,11 +222,16 @@ def main():
 
         # Some common attributes
         domain = to_text(challenge_data['resource'])
-        subject = issuer = cryptography.x509.Name([
-            cryptography.x509.NameAttribute(cryptography.x509.oid.NameOID.COMMON_NAME, domain),
-        ])
+        type, identifier = to_text(challenge_data.get('resource_original', 'dns:' + challenge_data['resource'])).split(':', 1)
```

Comment:
not a blocker, but I would rename `type` to `_type`, type is a reserved word in python

Reply:
Good point! I'll fix that in this PR and #53660.